main.cpp

```diff
@@ -77,7 +80,14 @@ struct U
     float update(float* v)      //12
     {
         std::cout << "U's a value: " << a << std::endl;
-        a = *v;
+        if( v != nullptr)
+        {
+            a = *v;
+        }
+        else 
+        {
+            std::cout << "error: v = nullptr" << std::endl;
+        }
```

Comment:
in the static function, you exit early when `v` is a nullptr.
you should use the same pattern here in this member function
```
 12) replicate the functionality of <structName2>'s static function by implementing a member function in U that does the same thing.
```
- [ ] revise